include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1078 @@
+                        outFile << "#step emit_all" << std::scientific;
+                        for (int i = startWindow_y; i < (endWindow_y + 10); i+=10)
+                        {
+                            outFile << " " << i * UNIT_LENGTH * SI::CELL_WIDTH_SI;
+                        }
+                        outFile << std::endl;
+                        fisttimestep = false;
+                    } 
+                    outFile << currentStep << " "
+                            << std::scientific;
+
+                    long double numElec_all = 0.0;
+                    long double ux2_all = 0.0;
+                    long double pos2_SI_all = 0.0;
+                    long double xux_all = 0.0;
+
+                    for (int i = startWindow_y; i < endWindow_y; i++)
+                    {
+                            numElec_all+= static_cast<long double>(globalCount_e.getDataPointer( )[i]);
+                            ux2_all    += static_cast<long double>(globalSumMom2.getDataPointer( )[i]) * UNIT_MASS* UNIT_MASS /(SI::ELECTRON_MASS_SI*SI::ELECTRON_MASS_SI);
+                            pos2_SI_all+= static_cast<long double>(globalSumPos2.getDataPointer( )[i]) *UNIT_LENGTH*UNIT_LENGTH ;
+                            xux_all    += static_cast<long double>(globalSumMomPos.getDataPointer( )[i])*UNIT_MASS *UNIT_LENGTH /SI::ELECTRON_MASS_SI;
+                    }
+                    float64 emit_all = algorithms::math::sqrt(((float64)pos2_SI_all * (float64)ux2_all - (float64)xux_all * (float64)xux_all)/((float64)numElec_all*(float64)numElec_all));
+                    if (emit_all > 0.0){
+                        outFile << emit_all << " ";
+                    }
+                    else {
+                        outFile << "0.0 ";
+                    }
+
+                    for (int i = startWindow_y; i < endWindow_y; i+=10)
+                    {
+                        double numElec = globalCount_e.getDataPointer( )[i];
+                        double mom2_SI   = globalSumMom2.getDataPointer( )[i] * UNIT_MASS * UNIT_SPEED* UNIT_MASS * UNIT_SPEED;
+                        double pos2_SI   = globalSumPos2.getDataPointer( )[i] *UNIT_LENGTH*UNIT_LENGTH ;
+                        double mompos_SI = globalSumMomPos.getDataPointer( )[i]*UNIT_MASS * UNIT_SPEED*UNIT_LENGTH;
+                        for ( int j=i+1; j<i+10 && j < endWindow_y; j++ ){
+                                numElec   += globalCount_e.getDataPointer( )[j];
+                                mom2_SI   += globalSumMom2.getDataPointer( )[j] * UNIT_MASS * UNIT_SPEED* UNIT_MASS * UNIT_SPEED;
+                                pos2_SI   += globalSumPos2.getDataPointer( )[j] *UNIT_LENGTH*UNIT_LENGTH;
+                                mompos_SI += globalSumMomPos.getDataPointer( )[j]*UNIT_MASS * UNIT_SPEED*UNIT_LENGTH;
+                        }
+                        float64 ux2 = mom2_SI/(UNIT_SPEED*UNIT_SPEED*SI::ELECTRON_MASS_SI*SI::ELECTRON_MASS_SI);
```

Comment:
my fault I said you should use `float64` but it must be `float_64` (sry) Could you please rename all types.

Reply:
is it right and complete like this?